### PR TITLE
#184759907 implement creating product

### PR DESCRIPTION
**Pull request by Abdden**
### What does this PR do?

Creating Product

### Description of Tasks to be Completed?

A seller should be able to create products

### How should this be manually tested?

- Clone [repo](https://github.com/atlp-rwanda/vikings-ec-fe.git)
- Change into the directory **vikings-ec-fe**
- Checkout to the branch: `feature/create-product-184759907`
- Run the command `npm install` 
- Run the command `npm run dev`
- Try to login as an admin [ email: `admin@gmail.com`, password: `Password@123` ]
- Got to `http://localhost:8000/dashboard/products`
- Click on add new product

### Testing the deployed preview

- Got to `https://vikings-ec-fe-git-feature-create-product-fddcd9-andela-vikings.vercel.app`
- Login as an admin [ email: `admin@gmail.com`, password: `Password@123` ]
- Add `/dashboard/products` in the URL and press Enter
- Click on add new product
- Fill out the necessary details in the form
- Click save

### What are the relevant pivotal tracker stories?
[#184759907](https://www.pivotaltracker.com/story/show/184759907)

### CODE REVIEW

**Review by irakozetony — APPROVED** [review]

**Review by Aline-Uwera — APPROVED** [review]

**Review by ndatumuremyi — APPROVED** [review]

**Review by irakozetony — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -109,6 +109,7 @@
     "url-loader": "^4.1.1",
     "webpack": "^5.76.2",
     "webpack-cli": "^5.0.1",
-    "webpack-dev-server": "^4.13.1"
+    "webpack-dev-server": "^4.13.1",
+    "yup": "^1.0.2"
```
Comment (ncutixavier) [inline package.json]: Why are you using two similar packages?

File: src/components/forms/CreateProductForm.js [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+    delete productData.images;
+    let date = productData.expiryDate;
+    delete productData.expiryDate;
+    date = convertDate(date);
+    const newData = getFormFromObject(productData);
+    selectedImages.forEach((image) => {
+      newData.append('images', image);
+    });
+    newData.append('expiryDate', date);
+    console.log(newData);
```
Comment (ncutixavier) [inline src/components/forms/CreateProductForm.js]: Remove all unnecessary logs

File: src/features/api/customAxios.js [hunk 3]
```diff
@@ -16,18 +16,19 @@ const responseHandler = (response) => response;
 const errorHandler = (error) => {
   if (error.response.status === 401) {
     localStorage.clear();
-    return (location.href = `${process.env.FRONT_END_URL}/auth/signin`);
+    return (location.href =
+      'https://vikings-ec-fe-git-feature-create-product-fddcd9-andela-vikings.vercel.app/auth/signin');
```
Comment (ncutixavier) [inline src/features/api/customAxios.js]: You don't have to return a full link, just redirect user to auth route.

Hint: `window.location.href='/auth'`

File: src/layout/CreateProductLayout.js
Comment (ncutixavier) [inline src/layout/CreateProductLayout.js:1]: - This is not a layout, 
- you can combine this file and create a product form

File: src/pages/seller/CreateProductPage.jsx [hunk 4]
```diff
@@ -0,0 +1,5 @@
+import React from 'react';
+import CreateProductForm from '../../components/forms/CreateProductForm';
+
+const CreateProductPage = () => <CreateProductForm />;
+export default CreateProductPage;
```
Comment (ncutixavier) [inline src/pages/seller/CreateProductPage.jsx]: This is not needed.

File: src/store.js [hunk 5]
```diff
@@ -41,12 +43,14 @@ const store = configureStore({
...
-    profile,
-    updateProfile,
     updatePassword: updatePasswordReducer,
+    createProduct: createProductReducer,
     product: getProductsReducer,
     singleProduct: singleProductReducer,
     getRatings: getRatingsReducer,
+    profile,
+    updateProfile,
+    category,
```
Comment (ncutixavier) [inline src/store.js]: Let's follow the same convention

**Review by ncutixavier — CHANGES_REQUESTED** [review]
Please disable your formatter because you're doing unnecessary changes

File: package.json (hunk 1 above)
Comment (Abdden) [reply package.json]: That is not the case.
In fact, it is a single package. I only added another version of the Yup package.

File: src/store.js [hunk 6]
```diff
@@ -41,12 +43,14 @@ const store = configureStore({
...
-    profile,
-    updateProfile,
     updatePassword: updatePasswordReducer,
+    createProduct: createProductReducer,
     product: getProductsReducer,
     singleProduct: singleProductReducer,
     getRatings: getRatingsReducer,
+    updateProfile: updateProfile,
+    profile: profile,
+    category: category,
```
Comment (ncutixavier) [inline src/store.js]: They are not supposed to have a similar name. Check above value and try to relate

**Review by ncutixavier — APPROVED** [review]

**Comment by ncutixavier** [comment]
- In the expiry date, disable the old dates
- After Succesful adding the product, clear the images section and redirect the user to his/her products pages

**Review by ncutixavier — APPROVED** [review]

**Pull request merged**